modules/xfem/src/base/XFEMFuncs.C

```diff
@@ -0,0 +1,597 @@
+
+    wss[2][0] = 8.5191465330460049e-01;
+    wss[2][1] = 4.5560372783619284e-01;
+    wss[2][2] = 5.6607220700753210e-01;
+
+    wss[3][0] = -wss[2][0];
+    wss[3][1] = wss[2][1];
+    wss[3][2] = wss[2][2];
+
+    wss[4][0] = 6.3091278897675402e-01;
+    wss[4][1] = -7.3162995157313452e-01;
+    wss[4][2] = 6.4271900178367668e-01;
+
+    wss[5][0] = -wss[4][0];
+    wss[5][1] = wss[4][1];
+    wss[5][2] = wss[4][2];
+  }
+  else
+    mooseError("Unknown Wissmann quadrature type");
+}
+
+void shapeFunc2D(unsigned int nen, std::vector<Real> &ss, std::vector<Point> &xl,
+                 std::vector<std::vector<Real> > &shp, Real &xsj, bool natl_flg)
+{
+  // Get shape functions and derivatives
+  Real s[4] = {-0.5, 0.5, 0.5, -0.5};
+  Real t[4] = {-0.5, -0.5, 0.5, 0.5};
+
+  if (nen == 4) // quad element
+  {
+    Real xs[2][2] = {{0.0,0.0},{0.0,0.0}};
+    Real sx[2][2] = {{0.0,0.0},{0.0,0.0}};
+    for (unsigned int i = 0; i < 4; ++i)
+    {
+      shp[i][2] = (0.5 + s[i]*ss[0])*(0.5 + t[i]*ss[1]);
+      shp[i][0] = s[i]*(0.5 + t[i]*ss[1]);
+      shp[i][1] = t[i]*(0.5 + s[i]*ss[0]);
+    }
+    for (unsigned int i = 0; i < 2; ++i) //x, y
+    {
+      for (unsigned int j = 0; j < 2; ++j) // xi, eta
+      {
+        xs[i][j] = 0.0;
+        for (unsigned int k = 0; k < nen; ++k)
+          xs[i][j] += xl[k](i)*shp[k][j];
+      }
+    }
+    xsj = xs[0][0]*xs[1][1] - xs[0][1]*xs[1][0]; // det(j)
+    if (natl_flg == false) // get global derivatives
+    {
+      Real temp = 1.0/xsj;
+      sx[0][0] =  xs[1][1]*temp; // inv(j)
+      sx[1][1] =  xs[0][0]*temp;
+      sx[0][1] = -xs[0][1]*temp;
+      sx[1][0] = -xs[1][0]*temp;
+      for (unsigned int i = 0; i < nen; ++i)
+      {
+        temp      = shp[i][0]*sx[0][0] + shp[i][1]*sx[1][0];
+        shp[i][1] = shp[i][0]*sx[0][1] + shp[i][1]*sx[1][1];
+        shp[i][0] = temp;
+      }
+    }
+  }
+  else if (nen == 3) // triangle element
+  {
+    // x1*(y2 - y3) + x2*(y3 - y1) + x3*(y1 - y2)
+    xsj  = xl[0](0)*(xl[1](1)-xl[2](1)) + xl[1](0)*(xl[2](1)-xl[0](1)) + xl[2](0)*(xl[0](1)-xl[1](1));
+    Real xsjr = 1.0;
+    if (xsj != 0.0)
+        xsjr = 1.0/xsj;
+    // xsj  *= 0.5; // we do not have this 0.5 here because in stdQuad2D the sum of all weights in tri is 0.5
+    shp[0][2] = ss[0];
+    shp[1][2] = ss[1];
+    shp[2][2] = ss[2];
+    if (natl_flg == false) // need global drivatives
+    {
+      shp[0][0] = (xl[1](1) - xl[2](1))*xsjr;
+      shp[0][1] = (xl[2](0) - xl[1](0))*xsjr;
+      shp[1][0] = (xl[2](1) - xl[0](1))*xsjr;
+      shp[1][1] = (xl[0](0) - xl[2](0))*xsjr;
+      shp[2][0] = (xl[0](1) - xl[1](1))*xsjr;
+      shp[2][1] = (xl[1](0) - xl[0](0))*xsjr;
+    }
+    else
+    {
+      shp[0][0] = 1.0;
+      shp[0][1] = 0.0;
+      shp[1][0] = 0.0;
+      shp[1][1] = 1.0;
+      shp[2][0] = -1.0;
+      shp[2][1] = -1.0;
+    }
+  }
+  else
+    mooseError("ShapeFunc2D only works for linear quads and tris!");
+}
+
+double r8vec_norm(int n, double a[])
```

Comment:
I agree with @dschwen on this. At the very least all of these functions should be put in a Namespace especially if you are hesitant to match our coding conventions. We do allow deviations from conventions for "contrib" codes. If you don't want to change it, how about putting them in contrib?

As far as wrappers go, it would be easy to have an XFem header that defines the wrapper now (just a few typedefs will do until you pull it out). You could define your own "Real" type too so it matches MOOSE. That may not be necessary if you put all of this in contrib though. I just don't like seeing it mixed in with the module code.